### PR TITLE
RB-4 Charge cards requests

### PR DESCRIPTION
**Pull request by stymoshchuk**

### CODE REVIEW

File: lib/unit/api_resources/card_resource.rb [hunk 1]
```diff
@@ -83,6 +83,12 @@ def get_card(card_id)
           response_handler(response)
         end
 
+        def update(request)
```
Comment (AsafPolakovski) [inline lib/unit/api_resources/card_resource.rb:86]: missing docs?

Comment (Yaellever) [reply lib/unit/api_resources/card_resource.rb:88]: @stymoshchuk

File: lib/unit/models/card/create_business_virtual_credit_card_request.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  module Card
+    # @param account_id [String]
+    # @param full_name [FullName]
+    # @param date_of_birth [Date]
+    # @param address [Address]
+    # @param phone [Phone] - optional
+    # @param email [String] - optional
+    # @param idempotency_key [String] - optional
+    # @param tags [Hash] - optional
+    # @param limits [Hash] - optional
```
Comment (nirmaoz) [inline lib/unit/models/card/create_business_virtual_credit_card_request.rb:15]: missing expiryDate param?

File: lib/unit/models/card/patch_business_virtual_card_request.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            "attributes": {
+              "address": address,
+              "phone": phone,
+              "email": email,
+              "tags": tags,
+              "limits": limits
+            }
+          }
+        }
+        payload[:data][:attributes].compact!
```
Comment (nirmaoz) [inline lib/unit/models/card/patch_business_virtual_card_request.rb:39]: please consider calling compact on attributes before setting them:

```ruby
"attributes": {
              "address": address,
              "phone": phone,
              "email": email,
              "tags": tags,
              "limits": limits
            }.compact
```

File: lib/unit/models/card/card.rb [hunk 4]
```diff
@@ -50,6 +56,94 @@ def create_business_debit_card(account_id:, full_name:, date_of_birth:, address:
...
+      # @param shipping_address [Address] - optional
+      # @param phone [Phone] - optional
+      # @param email [String] - optional
+      # @param design [String] - optional
+      # @param additional_embossed_text [String] - optional
+      # @param idempotency_key [String] - optional
+      # @param tags [Hash] - optional
+      # @param limits [Hash] - optional
+      # @param print_only_business_name [Boolean] - optional
+      def create_business_credit_card(account_id:, full_name:, date_of_birth:, address:, shipping_address: nil, phone: nil, email: nil, design: nil, additional_embossed_text: nil, idempotency_key: nil, tags: nil, limits: nil, print_only_business_name: nil)
```
Comment (nirmaoz) [inline lib/unit/models/card/card.rb:74]: missing expiryDate ?

File: lib/unit/models/card/create_business_debit_card_request.rb [hunk 5]
```diff
@@ -5,7 +5,7 @@
 module Unit
   module Card
     class CreateBusinessDebitCardRequest
-      attr_reader :account_id, :full_name, :date_of_birth, :address, :shipping_address, :phone, :email, :design, :additional_embossed_text, :idempotency_key, :tags, :limits, :print_only_business_name
+      attr_reader :account_id, :full_name, :date_of_birth, :address, :shipping_address, :phone, :email, :design, :additional_embossed_text, :idempotency_key, :tags, :limits, :print_only_business_name, :type
```
Comment (nirmaoz) [inline lib/unit/models/card/create_business_debit_card_request.rb:8]: missing expiryDate ?

**Review by nirmaoz — CHANGES_REQUESTED** [review]

**Review by nirmaoz — APPROVED** [review]

**Pull request merged**